crates/ruff_linter/src/rules/flake8_builtins/rules/stdlib_module_shadowing.rs

```diff
@@ -98,7 +98,11 @@ pub(crate) fn stdlib_module_shadowing(
 
     let module_name = components.next()?;
 
-    if is_allowed_module(settings, &module_name) {
+    if is_allowed_module(
+        settings,
+        settings.resolve_target_version(&path).minor,
```

Comment:
Could we pass in the target version here instead of resolving it over and over again for every file that's being linted?